lib/appmap.rb

```diff
@@ -112,19 +112,57 @@ def detect_metadata
   end
 end
 
-if Gem.loaded_specs['rails']
-  require 'active_support'
-  require 'active_support/core_ext'
-  require 'rails'
-  require 'appmap/railtie' 
-end
+lambda do
+  Initializer = Struct.new(:class_name, :module_name, :gem_module_name)
+
+  INITIALIZERS = {
+    'Rails::Railtie' => Initializer.new('AppMap::Railtie', 'appmap/railtie', 'railtie'),
+    'RSpec' => Initializer.new('AppMap::RSpec', 'appmap/rspec', 'rspec-core'),
+    'Minitest::Unit::TestCase' => Initializer.new('AppMap::Minitest', 'appmap/minitest', 'minitest')
+  }
+
+  print_msg = if defined?(::Rails) && Rails.logger
+    Rails.logger
+  elsif ENV['DEBUG'] == 'true'
+    Object.method(:warn)
+  else
+    ->(msg) {}
+  end
 
-if Gem.loaded_specs['rspec-core']
-  require 'appmap/rspec'
-end
+  TracePoint.new(:class) do |tp|
+    cls_name = tp.self.name
+    initializers = INITIALIZERS.delete(cls_name)
+    if initializers
+      initializers = [ initializers ] unless initializers.is_a?(Array)
+      next if Object.const_defined?(initializers.first.class_name)
+
+      gem_module_name = initializers.first.gem_module_name
+
+      print_msg.call AppMap::Util.color(<<~LOAD_MSG, :magenta)
+      When 'appmap' was loaded, '#{gem_module_name}' had not been loaded yet. Now '#{gem_module_name}' has
+      just been loaded, so the following AppMap modules will be automatically required:
+
+      #{initializers.map(&:module_name).join("\n")}
+
+      To suppress this message, require '#{gem_module_name}' before you require 'appmap' in your Gemfile.
```

Comment:
I think it's worth rewording this message to avoid the use of the word `require` -- because `require` is  Ruby keyword, and isn't actually used in a Gemfile, I thing this message may lead to confusion. You could say something like
```
ensure '#{gem_module_name}' appears before 'appmap' in your Gemfile.
```